src/shared/components/oncoprint/controls/OncoprintControls.tsx

```diff
@@ -568,9 +569,30 @@ export default class OncoprintControls extends React.Component<
         }));
     }
     
+    @autobind
+    @action onInputChange(input:string) {
+        if (input && this.props.treatmentSelectOptions) {
+            const lowerCaseInput = input.toLowerCase();
+            this.filteredTreatmentOptions = this.props.treatmentSelectOptions.filter(option => {
+                const lowerCaseLabel = option.label.toLowerCase();
+                const lowerCaseValue = option.value.toLowerCase();
+                return lowerCaseLabel.includes(lowerCaseInput) || lowerCaseValue.includes(lowerCaseInput);
+            });
+        } else {
+            this.filteredTreatmentOptions = this.props.treatmentSelectOptions;
+        }
+    }
+    
+    @autobind onAddAllTreatments() {
+        if (this.filteredTreatmentOptions) {
+            // merge the current selected options with all the filtered ones and remove duplicates
+            this.onSelectTreatments(_.uniqBy([...this.selectedTreatmentsJS, ...this.filteredTreatmentOptions], 'value'))
```

Comment:
we try to never use the lodash property shortcuts ("value") because they aren't type checked.  Always use thing=>thing.someProp

Reply:
I see. I have updated this line of code and added some logic to prevent clearing search input value after selecting an option to this PR. Please review it again. Thank you.